py-polars/src/series/mod.rs

```diff
@@ -281,9 +281,10 @@ impl PySeries {
     }
 
     fn series_equal(&self, other: &PySeries, null_equal: bool, strict: bool) -> bool {
-        if strict {
-            self.series.eq(&other.series)
-        } else if null_equal {
+        if strict && self.series.dtype().ne(other.series.dtype()) {
```

Comment:
Just a very nitpicky thing: The `!=` syntax is more readable to me.

```suggestion
        if strict && (self.series.dtype() != other.series.dtype()) {
```

Reply:
I'll go ahead and commit this so I can merge this and forget about it 😄